vehicles/src/modules/loa/loa.service.ts

```diff
@@ -0,0 +1,435 @@
+import {
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common';
+import { LogAsyncMethodExecution } from 'src/common/decorator/log-async-method-execution.decorator';
+import { CreateLoaDto } from './dto/request/create-loa.dto';
+import { IUserJWT } from 'src/common/interface/user-jwt.interface';
+import { ReadLoaDto } from './dto/response/read-loa.dto';
+import { InjectMapper } from '@automapper/nestjs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { LoaDetail } from './entities/loa-detail.entity';
+import { Brackets, DataSource, Repository } from 'typeorm';
+import { Mapper } from '@automapper/core';
+import { LoaVehicle } from './entities/loa-vehicles.entity';
+import { LoaPermitType } from './entities/loa-permit-type-details.entity';
+import { ReadFileDto } from '../common/dto/response/read-file.dto';
+import { DopsService } from '../common/dops.service';
+import { FileDownloadModes } from 'src/common/enum/file-download-modes.enum';
+import { Response } from 'express';
+import { UpdateLoaDto } from './dto/request/update-loa.dto';
+import { Nullable } from '../../common/types/common';
+
+@Injectable()
+export class LoaService {
+  private readonly logger = new Logger(LoaService.name);
+  constructor(
+    @InjectMapper() private readonly classMapper: Mapper,
+    @InjectRepository(LoaDetail)
+    private loaDetailRepository: Repository<LoaDetail>,
+    private dataSource: DataSource,
+    private readonly dopsService: DopsService,
+  ) {}
+
+  /**
+   * This method handles the creation of a LOA (Letter of Authorization).
+   *
+   * Steps:
+   * 1. Upload the file using the dopsService.
+   * 2. Map the createLoaDto to LoaDetail while injecting extra arguments such as companyId and documentId.
+   * 3. Set the isActive property of the LOA to true.
+   * 4. Save the LOA details in the repository.
+   * 5. Map the saved LOA details to ReadLoaDto.
+   * 6. Assign the fileName from readFileDto to ReadLoaDto and return it.
+   *
+   * @param {IUserJWT} currentUser - The current user making the request.
+   * @param {CreateLoaDto} createLoaDto - Data Transfer Object containing the data for creating the LOA.
+   * @param {number} companyId - ID of the company for which the LOA is being created.
+   * @param {Express.Multer.File} file - The file to be uploaded and associated with the LOA.
+   * @returns {Promise<ReadLoaDto>} - Returns a ReadLoaDto containing the details of the created LOA.
+   */
+  @LogAsyncMethodExecution()
+  async create(
+    currentUser: IUserJWT,
+    createLoaDto: CreateLoaDto,
+    companyId: number,
+    file: Express.Multer.File,
+  ): Promise<ReadLoaDto> {
+    const readFileDto = await this.dopsService.upload(
+      currentUser,
+      companyId,
+      file,
+    );
+
+    const loa = await this.classMapper.mapAsync(
```

Comment:
@gchauhan-aot Please ensure created user name, guid, updated user name  etc is persisted

Reply:
done